### PR TITLE
dts: Add binding and nodes representing Nordic ECB peripheral

### PR DESCRIPTION
**Pull request by anangl**
This PR adds binding for the Nordic ECB (AES electronic codebook mode encryption) peripheral and corresponding devicetree nodes for nRF SoCs that feature this peripheral. The `crypto_nrf_ecb` driver is converted so that it creates its instance basing on DT, and the `CRYPTO_NRF_ECB_NAME` Kconfig option is removed as no longer needed.

The first commit does an additional minor cleanup in one of the files modified by this PR.

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Comment by anangl** [comment]
Rebased and updated accordingly (used `NRF_DEFAULT_IRQ_PRIORITY` instead of `1` in `interrupts` properties) after #20894 has been merged.

**Review by mbolivar-nordic — APPROVED** [review]

File: drivers/crypto/Kconfig.nrf_ecb [hunk 1]
```diff
@@ -3,16 +3,12 @@
...
 
+# Workaround for not being able to have commas in macro arguments
+DT_COMPAT_NORDIC_NRF_ECB := nordic,nrf-ecb
 
-menuconfig CRYPTO_NRF_ECB
+config CRYPTO_NRF_ECB
 	bool "nRF AES electronic codebook mode encryption"
-	depends on HAS_HW_NRF_ECB && !BT_CTLR
+	depends on $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_ECB))
+	depends on !BT_CTLR
```
Comment (ioannisg) [inline drivers/crypto/Kconfig.nrf_ecb:12]: @anangl  should we have a comment explaining why !BT_CTRL is a dependency?

**Review by ioannisg — APPROVED** [review]
LGTM.
I left a suggestion for a comment to clarify why there's a dependency with the BT Controller

File: drivers/crypto/Kconfig.nrf_ecb (hunk 1 above)
Comment (ioannisg) [reply drivers/crypto/Kconfig.nrf_ecb:12]: I know it was un-commented, already, but perhaps we could add now the clarification.

Comment (anangl) [reply drivers/crypto/Kconfig.nrf_ecb:15]: @ioannisg I added a note that explains this dependency. Please check if it is acceptable.

**Comment by anangl** [comment]
Rebased on master, to solve the problem with reported undefined Kconfig options being referenced (those from the recently added `modules/audio/sof` project that was not fetched by west before the rebase).

File: drivers/crypto/Kconfig.nrf_ecb (hunk 1 above)
Comment (ioannisg) [reply drivers/crypto/Kconfig.nrf_ecb:15]: Looks fine, thanks

**Pull request merged**